optimum/habana/transformers/generation/utils.py

```diff
@@ -2029,7 +2046,16 @@ def _contrastive_search(
             ):
                 # Pad the returned pask key values tensors from prefill phase forward run to maximum length
                 # before starting the decode phase.
-                self._pad_past_key_values(model_kwargs)
+
+                is_mqa_model = self.config.model_type == 'gpt_bigcode' and self.config.multi_query
+                model_kwargs["mqa_model"] = is_mqa_model
+                if is_mqa_model:
```

Comment:
this ```if``` condition was removed recently by https://github.com/huggingface/optimum-habana/pull/1163 if anyone knows what the correct code should look like, please comment

Reply:
and same for line 2414, now there is a check in one place and no check in another place. Please advise if it should be aligned somehow https://github.com/huggingface/optimum-habana/pull/1218/files#diff-284d6c109e6788a4405e302113203f6b7e98be68afc7dd3d85b89c6329b9275eR2414